pkg/controller/machine.go

```diff
@@ -372,49 +372,53 @@ func (c *Controller) syncHandler(key string) error {
 
 // deleteMachineAndProviderInstance makes sure that an instance has gone in a series of steps.
 func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, machine *machinev1alpha1.Machine) error {
-	var err error
-	var providerInstance instance.Instance
 
-	// step 1: delete provider instance
-	if err = c.deleteProviderInstance(prov, machine); err != nil {
-		message := fmt.Sprintf("%v. Please manually delete finalizers from the machine object.", err)
-		return c.updateMachineErrorIfTerminalError(machine, machinev1alpha1.DeleteMachineError, message, err, "failed to delete machine at cloudprovider")
-	}
+	// step 1: get the provider instance.
+	providerInstance, err := c.getProviderInstance(prov, machine)
+	if err != nil {
+		// step 1.1: failed to get instance, because of some unknown error -> return and see if we need to handle a terminal error here
+		if err != cloudprovidererrors.ErrInstanceNotFound {
```

Comment:
some providers like `DO` detect the `ErrInsanceNotFound` and will return `nil` instead.

https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/digitalocean/provider.go#L350

Reply:
I'll update the do code, to return `cloudprovidererrors.ErrInstanceNotFound`